pulsar-client-cpp/include/pulsar/ProducerConfiguration.h

```diff
@@ -164,13 +204,49 @@ class PULSAR_PUBLIC ProducerConfiguration {
      */
     int getMaxPendingMessagesAcrossPartitions() const;
 
+    /**
+     * Set the message routing modes for partitioned topics.
+     *
+     * @param PartitionsRoutingMode partition routing mode.
+     * @return
+     */
     ProducerConfiguration& setPartitionsRoutingMode(const PartitionsRoutingMode& mode);
+
+    /**
+     * The getter associated with setPartitionsRoutingMode().
+     */
     PartitionsRoutingMode getPartitionsRoutingMode() const;
 
+    /**
+     * Set a custom message routing policy by passing an implementation of MessageRouter.
+     *
+     * @param messageRouter message router.
+     * @return
+     */
     ProducerConfiguration& setMessageRouter(const MessageRoutingPolicyPtr& router);
+
+    /**
+     * The getter associated with setMessageRouter().
+     */
     const MessageRoutingPolicyPtr& getMessageRouterPtr() const;
 
+    /**
+     * Set the hashing scheme, which is a standard hashing function available when choosing the partition to
+     * use for a particular message. <ul> <li>{@link HashingScheme#JavaStringHash}: Java {@code
+     * String.hashCode()} (Default) <li>{@link HashingScheme#BoostHash}: Use [Boost hashing
+     * function](https://www.boost.org/doc/libs/1_72_0/doc/html/boost/hash.html). <li>{@link
+     * HashingScheme#Murmur3_32Hash}: Use [Murmur3 hashing
+     * function](https://en.wikipedia.org/wiki/MurmurHash">https://en.wikipedia.org/wiki/MurmurHash).
+     * </ul>
```

Comment:
```suggestion
     * Set the hashing scheme, which is a standard hashing function available when choosing the partition to
     * use for a particular message.
     *
     * <p>Standard hashing functions available are:
     * <ul>
     * <li>{@link HashingScheme::JavaStringHash}: Java {@code String.hashCode()} (Default)
     * <li>{@link HashingScheme::BoostHash}: Use [Boost hashing
     * function](https://www.boost.org/doc/libs/1_72_0/doc/html/boost/hash.html).
     * <li>{@link HashingScheme::Murmur3_32Hash}: Use [Murmur3 hashing
     * function](https://en.wikipedia.org/wiki/MurmurHash")
     * </ul>
     *
```

Reply:
@BewareMyPower many thanks for your guidance! I've corporated ur comments, PTAL.